### PR TITLE
added `unload_current_scene` to SceneTree and removed ability for `change_scene_to` to continue when provided `null`

### PR DESCRIPTION
**Pull request by twobitadder**
This is the bigger half of the change mentioned in #63565 - this introduces a check for `null` on `change_scene_to` (it provides the same error of `ERR_CANT_CREATE` as if it were unable to instantiate the PackedScene). 

In order to provide the same functionality as that code, `unload_current_scene` was added which just nukes `current_scene`. Based on my working through the code, it seems like that was originally the path if `change_scene_to` was given a null value.

This is my first time interacting with the source code in anything more than a minor capacity, so comments and suggestions are absolutely welcome and encouraged, of course. I wouldn't be surprised if it could use some work to be in line with spec.

Thanks!

### CODE REVIEW

**Comment by twobitadder** [comment]
Case in point - just learned about `clang-format`. Revised files incoming.

**Comment by Diddykonga** [comment]
Personally I would prefer the naming `remove_current_scene()` instead, just because "unloading" sounds like it will remove the Scene from Load Cache, which it does not do, just a matter of semantics though.

**Comment by twobitadder** [comment]
I would be more inclined to agree if it passed back the scene, so that it would be more in line with behavior from `remove_child()`. As it stands though, this does at least attempt to `memdelete()` the object in question. If others agree with the name change though, I'm happy to do it.

File: scene/main/scene_tree.cpp [hunk 1]
```diff
@@ -1135,11 +1135,11 @@ Error SceneTree::change_scene(const String &p_path) {
 }
 
 Error SceneTree::change_scene_to(const Ref<PackedScene> &p_scene) {
+	ERR_FAIL_NULL_V(p_scene, ERR_CANT_CREATE);
+
 	Node *new_scene = nullptr;
```
Comment (AaronRecord) [inline scene/main/scene_tree.cpp:1140]:
```suggestion
	Node *new_scene = p_scene->instantiate();
```

Comment (twobitadder) [reply scene/main/scene_tree.cpp:1140]: This is reasonable - I didn't really take into consideration that the new flow wouldn't require that to be split up into two steps, haha.

**Comment by twobitadder** [comment]
Superseded by #71105 so i'll go ahead and close!

**Comment by YuriSizov** [comment]
Yes, sorry for that!

**Comment by twobitadder** [comment]
No problem at all! It's good to have a solution and it's hard to argue against one from reduz, haha

**Pull request closed without merging**